### PR TITLE
Cluster api updates

### PR DESCRIPTION
**Pull request by heytrav**
Changes to kubernetes documentation for upcoming ClusterAPI integration

### CODE REVIEW

**Comment by heytrav** [comment]
cc @dalees

File: source/kubernetes/versions.rst [hunk 1]
```diff
@@ -76,8 +76,8 @@ out of support.
 Catalyst Cloud Kubernetes Service supports the latest stable patch versions
 for each minor version. As long as there is a patch version released, the oldest
 patch version will be hidden/removed. For example, if current versions
-supported for v1.22.x are v1.22.17, then v1.22.17 will be hidden/removed in
-favor of the release of v1.22.18.
+supported for v1.30.x are v1.30.15, then v1.22.15 will be hidden/removed in
```
Comment (dalees) [inline source/kubernetes/versions.rst:79]:
```suggestion
supported for v1.30.x are v1.30.15, then v1.30.15 will be hidden/removed in
```

File: source/kubernetes/versions.rst [hunk 2]
```diff
@@ -122,9 +122,9 @@ Upgrading Kubernetes Version
 ****************************
 
 When doing Kubernetes version upgrade, minor version cannot be skipped. For
-example, if the current cluster version is v1.21.x, then it's not allowed
-to upgrade directly to v1.23.x. You have to upgrade to v1.22.x and then do another
-upgrade to v1.23.x.
+example, if the current cluster version is v1.29.x, then it's not allowed
+to upgrade directly to v1.31.x. You have to upgrade to v1.30.x and then do another
+upgrade to v1.31.x.
```
Comment (dalees) [inline source/kubernetes/versions.rst:127]: Probably worth linking to the Kubernetes skew policy here, and stating upgrades are in line with this.

**Review by dalees — CHANGES_REQUESTED** [review]

File: source/kubernetes/quickstart.rst [hunk 3]
```diff
@@ -28,11 +28,24 @@ used to create a production ready cluster.
 Pre-requisites
 **************
 
-Ensure user has the required privileges
+Service user
+============
+
+As a first step, you will need to create a service user in your account which you
```
[the remark this answers is not in the document]
Comment (heytrav) [reply source/kubernetes/quickstart.rst:34]: Yeah I had _We highly recommend_ in there before but then took it out. I was going to put a more detailed description in another part of the instructions. I wasn't sure if the Quickstart was a good point to digress into that.

File: source/kubernetes/quickstart.rst [hunk 4]
```diff
@@ -28,11 +28,24 @@ used to create a production ready cluster.
...
+then use to create Kubernetes clusters. The service user can have an arbitrary name 
+such as `preprod-serviceuser@myteam-myproject`. 
+
+
+Ensure service user has the required privileges
 =======================================
 
-In order to create a Kubernetes cluster you need to ensure the user has been
-allocated the ``heat_stack_owner`` role.
+In order to create a Kubernetes cluster the service user should only have the ``_member_``
```
[the remark this answers is not in the document]
Comment (heytrav) [reply source/kubernetes/quickstart.rst:42]: ok that's good to know. Although do we want to steer them away from adding things like `project_admin` and `k8s_admin` to that user?

[the remark this answers is not in the document]
Comment (heytrav) [reply source/kubernetes/quickstart.rst]: I actually put a Note in that explains that

File: source/kubernetes/quickstart.rst [hunk 5]
```diff
@@ -237,7 +260,7 @@ created using the dashboard method.
 
 .. Note::
 
-  Templates in the v1.20.x series onward need to specify the additional label:
+  Templates in the v1.30.x series onward need to specify the additional label:
```
Comment (heytrav) [inline source/kubernetes/quickstart.rst:263]: global search and replace here. Probably just drop this line.

File: source/kubernetes/quickstart.rst [hunk 6]
```diff
@@ -113,33 +128,36 @@ The following command will list all cluster templates available:
...
+``kubernetes-v1.29.6-dev-20200615``).
 
 Alternatively, a list of cluster templates can be seen in the
 **Cluster Template** dropdown of the **Create New Cluster** dialogue in the
 dashboard, under the **Container Infra** section.
 
 .. Note::
 
-  Templates that are from the v1.20.x range onwards use containerd at runtime to
+  Templates that are from the v1.30.x range onwards use containerd at runtime to
```
Comment (dalees) [inline source/kubernetes/quickstart.rst:150]: We should just update all docs to talk about containerd, now that docker is gone from all supported versions.

File: source/kubernetes/quickstart.rst (hunk 5 above)
Comment (dalees) [reply source/kubernetes/quickstart.rst:263]: Yeah - now that 1.20 is long past, we should just remove the version and state that all templates need this if you want to create a publicly accessibly K8s API.

**Comment by dalees** [comment]
Closing in favour of https://github.com/catalyst-cloud/catalystcloud-docs/pull/474

**Pull request closed without merging**